### ISSUE TITLE
When joining a room, it should be in the room list

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#15451]
given how long joins can take, the mid-join room should be in your room list so you can come back to it later, and only disappear if you explicitly remove it after the join fails/succeeds

**Comment by gsouquet** [comment]
Me and @jryans have discussed how to improve things. Before investing more time it would be great to have a bit more insights from @vector-im/design 

Here are some proposed changes to improve the UX when joining a room is a bit slower than expected

* If a joining a room takes more than 2 seconds, add a "mock" room tile in the left sidebar. This is to avoid having a flickering UI 
* Updating the joining room view to automatically peek in the room if possible
* Updating the joining room view and add an explainer as to why joining a room can be slow. It would probably be a plus to display that message only after a certain time to avoid over share technical details when this is not needed

The mock tiles would only consist of: 
* Room name (or defaults back to room id / alias)
* Avatar URL

The correct tag can be inferred from the `EventContent#is_direct` flag for invites, and if a user joins a public room, we can safely assume it will not be a DM

**Comment by t3chguy** [comment]
> or defaults back to room id

A room id is normally *never* shown in the room list, it'll say `Empty Room` instead if a room has no other factors to generate a name from https://matrix.org/docs/spec/client_server/r0.6.1#calculating-the-display-name-for-a-room

> The correct tag can be inferred 

You still don't know whether Favs/Low priority
And what space to show it under

**Comment by gsouquet** [comment]
> > The correct tag can be inferred
> 
> You still don't know whether Favs/Low priority (given its a newly joined room they'll likely not be in these, but I doubt Synapse clears your room tags if you leave & rejoin)

Thank you for the heads up on that one. I didn't know that synapse would keep a reference to your old room tags if you left and joined again. I'll test that one as it definitely could be an issue. My biggest fear with a mocked tile is to display it somewhere different than it would get displayed if the information was returned by synapse

> And what space to show it under (so after joining it might vanish from the room list/the space might switch from under you)

When accepting an invite or joining a public suggested room I was thinking of storing a reference to the current space identifier and only showing that mock when viewing that same space

**Comment by t3chguy** [comment]
in the case of a suggested room in a space it already all works in the room list
in the case of joining a public room from the room directory on the Home space, if its contained in one of the spaces you are in, as soon as the join is processed it'll jump from the Home space to one of the spaces which includes that room.
